addons/nuxeo-template-rendering/nuxeo-template-rendering-core/src/main/java/org/nuxeo/template/serializer/executors/XMLSerializer.java

```diff
@@ -0,0 +1,207 @@
+
+/**
+ * {@link TemplateInput} parameters are stored in the {@link DocumentModel} as a single String Property via XML
+ * Serialization. This class contains the Serialization/Deserialization logic.
+ *
+ * @author Tiry (tdelprat@nuxeo.com)
+ * @author bjalon (bjalon@qastia.com)
+ */
+public class XMLSerializer implements Serializer {
+
+    protected static final Log log = LogFactory.getLog(XMLSerializer.class);
+
+    public static final String XML_NAMESPACE = "http://www.nuxeo.org/DocumentTemplate";
+
+    public static final String XML_NAMESPACE_PREFIX = "nxdt";
+
+    public static final Namespace ns = new Namespace(XML_NAMESPACE_PREFIX, XML_NAMESPACE);
+
+    public static final QName fieldsTag = DocumentFactory.getInstance().createQName("templateParams", ns);
+
+    public static final QName fieldTag = DocumentFactory.getInstance().createQName("field", ns);
+
+    public static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss.SSS";
+
+    @Override
+    public List<TemplateInput> doDeserialization(String xml) throws DocumentException {
+        List<TemplateInput> result = new ArrayList<>();
+
+        Document xmlDoc = DocumentHelper.parseText(xml);
+
+        List<Element> nodes = xmlDoc.getRootElement().elements(fieldTag);
+
+        for (Element node : nodes) {
+            result.add(extractTemplateInputFromXMLNode(node));
+        }
+
+        return result;
+    }
+
+    protected TemplateInput extractTemplateInputFromXMLNode(Element node) throws DocumentException {
+        String paramName = getNameFromXMLNode(node);
+        InputType paramType = getTypeFromXMLNode(node);
+        String paramDesc = node.getText();
+        Boolean isReadonly = getIsReadonlyFromXMLNode(node);
+        Boolean isAutoloop = getIsAutoloopFromXMLNode(node);
+
+        Object paramValue = node.attributeValue("value");
+        switch (paramType) {
+            case StringValue:
+            case BooleanValue:
+                break;
+            case DateValue:
+                try {
+                    SimpleDateFormat dateFormat = new SimpleDateFormat(DATE_FORMAT);
+                    paramValue = dateFormat.parse((String) paramValue);
+                } catch (ParseException e) {
+                    throw new DocumentException(e);
+                }
+                break;
+           case MapValue:
+                Map<String, TemplateInput> listValue = new HashMap<>();
+                for (Element childNode : node.elements()) {
+                    TemplateInput childParam = extractTemplateInputFromXMLNode(childNode);
+                    if (childNode != null) {
+                        listValue.put(childParam.getName(), childParam);
+                    }
+                }
+                paramValue = listValue;
+                break;
+            case MapValue:
+            case ListValue:
+                LinkedHashMap<String, TemplateInput> listValue = new LinkedHashMap<>();
```

Comment:
No because order is important for the map. If the templateData is defined like that :
```
<map>
  <entry key="key1" value="value1"/>
  <entry key="key3" value="value3"/>
  <entry key="key2" value="value2"/>
</map>
```
or like that :
```
<list>
  <entry key="key1" value="value1"/>
  <entry key="key3" value="value3"/>
  <entry key="key2" value="value2"/>
</list>
```
User would like that when he will iterate through the list or the map the order will be the same.
(Think that case above was not the right one - see comment above).

Reply:
@kevinleturc  ? Are you agree with my comment ?

Reply:
I'm 👍